### PR TITLE
Interactive features

### PR DESCRIPTION
**Pull request by Armynerh**
Added a new JavaScript file and imported it as a module:
it contains methods related to the status updates (completed: true / false).
Added event listener to the checkbox (change).
Updated items object's value for completed key upon user actions.
Implemented a function for the "Clear all completed" button (use filter() method).
Stored the updated array of items in local storage, so the user gets the correct list of values after the page reloads (which means that any changes made to the list should be preserved).

### CODE REVIEW

File: src/completed.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  });
+  localStorage.setItem('todos', JSON.stringify(todos));
+  refresh();
+};
+export const clearCompleted = () => {
+  let todos = getTodos();
+  todos = todos.filter((todo) => !todo.completed);
+  localStorage.setItem('todos', JSON.stringify(todos));
+  refresh();
+};
```
Comment (usorfaitheloho) [inline src/completed.js]: You did well in implementing the  `interactive` feature, however, there seems to be an abnormal behavior in your app such that when I click on an item to be `cleared`, it selects two items which in turn would be deleted. Please this isn't the expected behavior and I'd kindly suggest you fix it 😄 

image

**Review by usorfaitheloho — CHANGES_REQUESTED** [review]
> Hi @Armynerh  ,
> 
> Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

  [ ](https://camo.githubusercontent.com/679757134a8fee76cfe1941f9f3ecb0e0cf56dd02da4498268ce31f0dacdfca3/68747470733a2f2f6d65646961302e67697068792e636f6d2f6d656469612f556461535559436c723932636a57706556782f67697068792e676966)
> 
> ## Highlights
> * All linter checks passed ✔️
> * Descriptive pull request ✔️
> * Used GitHub flow ✔️
> 
> ## Required Changes ♻️
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Nemwel-Boniface — COMMENTED** [review]
## Hello @Armynerh   👋,

I am Nemwel your reviewer.


## _This is an additional review as a result of the second opinion request._

I have received your second opinion request with a complaint about the previous reviewer @usorfaitheloho  Below are my findings:

When I inspected the project at first I was not able to spot the issue that @usorfaitheloho  mentioned and I was almost ignored it. However on closer inspection, I created a few todos and I deleted some in random order I then added a few other todos and when I was selecting them in random order, I realized that the second last when checked automatically also checks the last todo and vice versa. This is not the desired functionality as this is mainly a logical flaw in your code somewhere.

[Screencast from 22-07-2023 12:29:25 ASUBUHI.webm](https://github.com/Armynerh/todo-list-project/assets/86318284/b5c9bd2c-48bb-4437-8d38-9761ba4cac91)

**Steps I took to trouble shoot:**
1. I started by checking whether your delete completed todos works as expected by making sure that when I delete the finished tasks, it updates the count of the remaining todos. This seemed to be ok so it was not the issue :heavy_check_mark: 
2. I then moved to the create a todo and realized that the issue is that the index of the new todos is not being set correctly when you add them after deleting some todos. To fix this, you need to ensure that the index of the new todos is assigned correctly based on the current number of todos present.:bulb:

I made some slight updates to your ```addTodo``` func which I will be kind enough to link below. 

```
export const addTodo = (text) => {
  const todos = getTodos();
  const maxIndex = todos.reduce((max, todo) => (todo.index > max ? todo.index : max), -1);
  const todo = {
    description: text,
    completed: false,
    index: maxIndex + 1, // Assigning the correct index for the new todo
  };
  todos.push(todo);
  localStorage.setItem('todos', JSON.stringify(todos));
  refresh();
};
```

Implementing the above seemed to resolve the issue correctly and the index was being updated correctly and also the issue that @usorfaitheloho  faced seemed resolved.

**What to do:**

- [ ] Kindly ensure that you resolve the issue raised by @usorfaitheloho  and when you are done feel free to request for another review in your dashboard. :rocket:  


In case anything is not 100% clear tag me @Nemwel-Boniface so that I get the notification.

Keep rocking 🚀!

**Comment by Armynerh** [comment]
@Nemwel-Boniface Thank you so much for your review, I really appreciate that you took the time to go through the problem and provide me with such a detailed response. I will fix this error and request another review. Thanks a lot.

**Review by Munsa1 — APPROVED** [review]
Hi @Armynerh, 
Approved 1

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes :heavy_check_mark: 
App Works As Expected :heavy_check_mark: 

## Optional suggestions

- [ ] Consider deploying your app and adding the link to the `Live Demo` section of your readme file, or consider removing the live demo section from your readme file if you choose to not deploy. This will make the readme file more professional.


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**